src/Form/ConstraintForm.php

```diff
@@ -0,0 +1,233 @@
+      $options[$definition['id']] = $definition['name'];
+    }
+
+    $form['policy'] = [
+      '#type' => 'hidden',
+      '#value' => $this->getRequest()->get('password_enhancements_policy')->id(),
+    ];
+
+    $settings_wrapper_id = 'ajax-wrapper';
+    $form['type'] = [
+      '#type' => 'select',
+      '#title' => $this->t('Constraint'),
+      '#description' => $this->t('The type of the constraint.'),
+      '#options' => $options,
+      '#default_value' => $this->entity->getType(),
+      '#required' => TRUE,
+      '#ajax' => [
+        'callback' => '::updateForm',
+        'event' => 'change',
+        'wrapper' => $settings_wrapper_id,
+      ],
+      '#disabled' => !$this->entity->isNew(),
+    ];
+
+    $form['wrapper'] = [
+      '#type' => 'container',
+      '#prefix' => "<div id='{$settings_wrapper_id}'>",
+      '#suffix' => '</div>',
+      'descriptionSingular' => [
+        '#type' => 'textfield',
+        '#title' => $this->t('Description (singular)'),
+        '#description' => $this->t('The singular description that should be displayed for the password field.'),
+        '#required' => TRUE,
+      ],
+      'descriptionPlural' => [
+        '#type' => 'textfield',
+        '#title' => $this->t('Description (plural)'),
+        '#description' => $this->t('The plural description that should be displayed for the password field.'),
+        '#required' => TRUE,
+      ],
+      'settings' => [
+        '#type' => 'fieldset',
+        '#title' => $this->t('Constraint settings'),
+        '#tree' => TRUE,
+      ],
+    ];
+
+    try {
+      $form['wrapper'] = static::updateForm($form, $form_state);
+    }
+    catch (PluginException $e) {
+      $this->messenger->addError($this->t('Invalid plugin ID: @error', [
+        '@error' => $e->getMessage(),
+      ]));
+      watchdog_exception('password_enhancements', $e);
+    }
+
+    return $form;
+  }
+
+  /**
+   * Updates settings from.
+   *
+   * @param array $form
+   *   The form.
+   * @param \Drupal\Core\Form\FormStateInterface $form_state
+   *   The from's current state.
+   *
+   * @return array
+   *   Renderable array for the settings wrapper.
+   *
+   * @throws \Drupal\Component\Plugin\Exception\PluginException
+   */
+  public function updateForm(array $form, FormStateInterface $form_state): array {
+    // Get type if exist.
+    if (($type = $this->entity->getType()) === NULL) {
+      $type = $form_state->getValue('type');
+    }
+
+    if ($type !== NULL) {
+      // Get plugin instance for the selected type.
+      $plugin_instance = $this->passwordConstraintPluginManager->createInstance($type, $this->entity->getConfiguration());
+
+      // Update description.
+      if ($this->entity->isNew()) {
+        // Unset description from the form state if the type was changed.
+        $input = &$form_state->getUserInput();
+        if (array_key_exists('descriptionSingular', $input)) {
+          unset($input['descriptionSingular']);
```

Comment:
:roll_eyes: Could not you use AjaxCommands to update form values?
https://www.drupal.org/docs/8/api/ajax-api/core-ajax-callback-commands

Reply:
Not sure how, although I have to update all of the contents in the wrapper, so I guess this is the easiest way to do it.

Reply:
That could be done by Ajax callback commands as well. If you do not want to investigate AJAX callback commands now, please create a follow-up issue in this Github repo for later.

Reply:
I've created a ticket for this.